### PR TITLE
🐛storage: fixes query to file metadata

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- 🐛 Fixes bug introduced with [this change](https://github.com/ITISFoundation/osparc-simcore/pull/5796/files#diff-0e92b3dc5f2ea70b28592e67fd63e75c9767cfb1c3863d12b2a8956f8d8d2366) in the database query
- 🐛 Fixes access rights to storage data (in coordination with @matusdrobuliak66 )

## Related issue/s

- resolves ITISFoundation/osparc-issues#1587

## How to test

- The changes were **driven** by to **tests** in `storage`, namely
   - Test to check correctness of sql expression (also to print an manually test against master db)
       -  `pytest -vv tests/unit -k test_building_sql_statements`
   - Test to checks different queries in `GET locations/{loc_id}/files/metadata`
       - `pytest -vv tests/unit -k test_get_files_metadata`
- **Manual exploratory testing** using steps in  ITISFoundation/osparc-issues#1587 on a local deploy

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/storage/tests/unit/test_handlers_files_metadata.py [hunk 1]
```diff
@@ -48,29 +64,52 @@ async def test_get_files_metadata(
...
+    await create_project_access_rights(
+        project_id=project_id,
+        user_id=other_user_id,
+        read=True,
+        write=True,
+        delete=True,
+    )
+    response = await client.get(
+        f"{url.update_query(project_id=str(project_id), user_id=other_user_id)}"
+    )
```
Comment (sanderegg) [inline services/storage/tests/unit/test_handlers_files_metadata.py:89]: thanks!

**Review by sanderegg — COMMENTED** [review]
cool thanks. not sure if that is ready for review. if yes you have my approval

File: services/storage/src/simcore_service_storage/db_access_layer.py [hunk 2]
```diff
@@ -154,6 +155,24 @@ async def list_projects_access_rights(
...
+        .where(user_to_groups.c.uid == f"{user_id}")
+        .group_by(user_to_groups.c.uid)
+    )
+
+    return sa.select(projects.c.prj_owner, projects.c.access_rights).where(
+        (projects.c.uuid == f"{project_id}")
+        & (
+            (projects.c.prj_owner == f"{user_id}")
+            | sa.func.jsonb_exists_any(
+                projects.c.access_rights, user_gids.scalar_subquery()
```
Comment (matusdrobuliak66) [inline services/storage/src/simcore_service_storage/db_access_layer.py:170]: project.c.access_rights is already redundant and should not be used after this refactoring: https://github.com/ITISFoundation/osparc-simcore/pull/6060 . (I have kept the column there to be extra safe and if somethings goes wrong we have still the data/structure there as it was before, my plan was to remove it after it goes to production in next sprint). Now we need to use `project_to_groups`, ex.: 

```
            access_rights_subquery = (
                sa.select(
                    project_to_groups.c.project_uuid,
                    sa.func.jsonb_object_agg(
                        project_to_groups.c.gid,
                        sa.func.jsonb_build_object(
                            "read",
                            project_to_groups.c.read,
                            "write",
                            project_to_groups.c.write,
                            "delete",
                            project_to_groups.c.delete,
                        ),
                    ).label("access_rights"),
                ).group_by(project_to_groups.c.project_uuid)
            ).subquery("access_rights_subquery")
```
as used in the simcore_service_webserver/projects/db.py

Yes indeed this is bug from my side I didn't notice the usage of project access rights with these `sa.DDL` statements here. I double check and this should be the last place where it is used.

Comment (pcrespov) [reply services/storage/src/simcore_service_storage/db_access_layer.py:171]: @matusdrobuliak66 takes over and will contribute with some changes to the PR

File: services/storage/tests/unit/test_db_access_layer_sql.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from faker import Faker
+from simcore_postgres_database.utils import as_postgres_sql_query_str
+from simcore_service_storage.db_access_layer import _get_project_access_rights_stmt
+
+
+def test_build_access_rights_sql_statements(faker: Faker):
+    def _check(func_smt, **kwargs):
+        print()
+        print(f"{func_smt.__name__:*^100}")
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/storage/tests/unit/test_db_access_layer_sql.py:15]: @sanderegg some answers:
- tests correctness of sql queries. Sometimes with different parameters. Note that sometimes queries are composed with many if-conditions based on input paramters
- I find the print handy to test manually against master database. I set real parameters (e.g. user_id) and manually explore variants. It is very handy
- The CI is useful for the first part. I agree that the prints are not very useful for the CI. The even pollute it. I will turn them into logs.debug

File: services/storage/src/simcore_service_storage/db_access_layer.py [hunk 4]
```diff
@@ -112,31 +114,58 @@ def _aggregate_access_rights(
...
 
+def assemble_array_groups(user_group_ids: list[GroupID]) -> str:
+    return (
+        "array[]::text[]"
+        if len(user_group_ids) == 0
+        else f"""array[{', '.join(f"'{group_id}'" for group_id in user_group_ids)}]"""
+    )
+
+
+access_rights_subquery = (
```
Comment (sanderegg) [inline services/storage/src/simcore_service_storage/db_access_layer.py:125]: Q: why is this in global scope? or should it not be a capitalized Final constant?

**Review by sanderegg — APPROVED** [review]
Thanks!, please double check the query declaration at global scope

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks! looking good.

File: services/storage/src/simcore_service_storage/db_access_layer.py (hunk 4 above)
Comment (matusdrobuliak66) [reply services/storage/src/simcore_service_storage/db_access_layer.py:125]: I guess thats on me, if you look a bit bellow it is reused in 2 different places.

**Pull request merged**